### ISSUE TITLE
insecure_skip_verify doesn't seem to work

### ISSUE THREAD

**Issue by ncoonrod** [rancher/rke2#1887]
**Environmental Info:**
RKE2 Version: v1.20.7+rke2r2

Node(s) CPU architecture, OS, and Version: Linux testuser 5.11.0-36-generic #40~20.04.1-Ubuntu SMP Sat Sep 18 02:14:19 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

Cluster Configuration: 3 servers

I'm trying to have my cluster pull images from a registry that is utilizing a self-signed certificate. In the file `/etc/rancher/rke2/registries.yaml` I put:
```
mirrors:
  registry.example.com:
    endpoint:
      - "https://registry.example.com"
configs:
  "registry.example.com":
    tls:
      insecure_skip_verify: true
```

Then I restart the servers using `sudo service rke2-server restart`. I've also tried restarting the entire machines.

In Rancher, when I view my workload trying to pull the image I see
```
Failed to pull image "registry.example.com/derp.com/herp:1.0.0": rpc error: code = Unknown desc = failed to pull and unpack image "registry.example.com/derp.com/herp:1.0.0": failed to resolve reference "registry.example.com/derp.com/herp:1.0.0": failed to do request: Head "https://registry.example.com/v2/derp.com/herp/manifests/1.0.0": x509: certificate is valid for ingress.local, not registry.example.com
```

**Steps To Reproduce:**

- Installed RKE2: 

**Expected behavior:**

**Actual behavior:**

**Additional context / logs:**

**Comment by brandond** [comment]
I'm not aware of any issues with that not working, and I make personal use of this functionality as well. Can you confirm that your file is formatted properly? Note that you don't need a `mirrors` section since the explicit endpoint you've listed is the same as the default for a registry named `registry.example.com`.

This should work:
```yaml
configs:
  registry.example.com:
    tls:
      insecure_skip_verify: true
```

**Comment by ncoonrod** [comment]
I changed my registries.yaml file to be like yours. This is my `/etc/rancher/rke2/registries.yaml` file:
```
configs:
  registry.example.com:
    tls:
      insecure_skip_verify: true
```

Restarted rke2 server, and still having issues.

I'm using the Harbor app that I installed through Rancher. I can push to the Harbor registry, and sign into Harbor no problem. It's just pulling images I can't do.

**Comment by brandond** [comment]
Your registry is literally named registry.example.com?

**Comment by ncoonrod** [comment]
The real domain name is super simple, very similar to `registry.example.com`. There is no port or path after it.

I've also tried:
```
configs:
  "registry.example.com:443":
    tls:
      insecure_skip_verify: true
```

Again, this is using the Harbor registry app, and I'm not quite sure what the issue is.

**Comment by brandond** [comment]
Just to be clear, this config file is only honored by the containerd built in to rke2, for pulling images to be run by rke2's kubelet. If there are things deployed within the cluster, such as a registry mirror, you would have to explicitly configure those apps to get similar behavior if they support it.

**Comment by ncoonrod** [comment]
Found the issue. The issue is related to containerd 1.4.9, possibly [issue #3882 specifically](https://github.com/containerd/containerd/issues/3882). I'm using several GPUs within the cluster and in `/etc/rancher/rke2/config.yaml`, I set `container-runtime-endpoint: unix:///run/containerd/containerd.sock`.

If I use `ctr image pull` to try and pull from the repo, I'm unable to. I'll try to fix this when I get the chance.

Sorry for bothering you guys, and thanks for the quick responses.
